### PR TITLE
feat(perf): scripts to create test data

### PR DESCRIPTION
**Pull request by RealOrangeOne**
### What is the context of this PR?

We need a way to create test data in bulk for performance testing etc.

https://officefornationalstatistics.atlassian.net/browse/CMS-948

Data creation is seeded, meaning it can be recreated reliably.

Data is identified by a prefix. To save complexity, it's only possible to delete all test data, rather than specific seeds.

### How to review

There are 3 new management commands. 

- `create_test_data` - creates data based on a seed.
- `delete_test_data` - removes all test data from the system (based on the prefix)
- `show_default_test_data_config` - displays the default test data generation config in the terminal for guidance

### Follow-up Actions

- Expand to cover more model types

### CODE REVIEW

**Review by MaciekBaron — CHANGES_REQUESTED** [review]

File: cms/test_data/config.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        return self
+
+
+class ModelCreationConfig(BaseModel):
+    count: NonNegativeInt | RangeConfig = 1
+
+
+class PageCreationConfig(ModelCreationConfig):
+    published: FractionalFloat = 0.5
+    revisions: NonNegativeInt | RangeConfig = 0
```
[the remark this answers is not in the document]
Comment (RealOrangeOne) [reply cms/test_data/config.py]: TIL. Thanks for looking into this. Yeah I agree this should be tweaked.

[the remark this answers is not in the document]
Comment (MaciekBaron) [reply cms/test_data/config.py]: But that's precisely my point: when using the CMS you can't create a page with no revisions.

File: cms/test_data/management/commands/create_test_data.py
Comment (kevinhowbrook) [inline cms/test_data/management/commands/create_test_data.py:1]: **Suggestion**

Would be nice to have the commands in the `make` file

File: cms/test_data/management/commands/delete_test_data.py
Comment (kevinhowbrook) [inline cms/test_data/management/commands/delete_test_data.py:1]: **Suggestion**

Would be nice to have the commands in the make file

File: cms/test_data/management/commands/create_test_data.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+            for _ in range(self.get_config_count(self.config.topics.revisions)):
+                topic_page.specific.save_revision()
+
+            if (
+                not topic_page.live
+                and topic_page.latest_revision_id
+                and self.faker.boolean(int(self.config.topics.published_probability * 100))
+            ):
+                topic_page.specific.latest_revision.publish()
```
Comment (kevinhowbrook) [inline cms/test_data/management/commands/create_test_data.py]: **Issue**

Does this actually execute? TopicPageFactory inherits from wagtail_factories.PageFactory, which creates pages as live=True by default. So not topic_page.live is always False

File: cms/test_data/tests/test_commands.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+from django.test import TestCase
+from django.test.testcases import SimpleTestCase
+from modelsearch.index import class_is_indexed
+from wagtail.models import ReferenceIndex
+
+from cms.datasets.models import Dataset
+from cms.images.models import CustomImage
+from cms.taxonomy.models import Topic
+from cms.test_data.config import TestDataConfig
+from cms.test_data.management.commands.create_test_data import SEEDED_DATA_PREFIX
```
Comment (kevinhowbrook) [inline cms/test_data/tests/test_commands.py]: **Issue**

Looks like this is in `cms.test_data.constants`  now

**Review by kevinhowbrook — CHANGES_REQUESTED** [review]
Had to merge in `main` as best as I could locally to test, but both commands work well. It would be nice to have the commands in the Makefile, nice one @RealOrangeOne 

I think the larger task here is rebasing or merging main but I'd love to see this go in then we can build it out further if we need/want to

File: cms/test_data/management/commands/create_test_data.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        parser.add_argument(
+            "--seed", nargs="?", default=4, type=int, help="Random seed to produce deterministic output"
+        )
+        parser.add_argument("--config", type=validate_config_file, default=TestDataConfig(), help="Config file")
+        parser.add_argument(
+            "--noinput",
+            "--no-input",
+            action="store_false",
+            dest="interactive",
+            help="Tells Django to NOT prompt the user for input of any kind.",
```
Comment (kevinhowbrook) [inline cms/test_data/management/commands/create_test_data.py:171]:
```suggestion
            help="Tells Django NOT to prompt the user for input of any kind.",
```

Comment (RealOrangeOne) [reply cms/test_data/management/commands/create_test_data.py:171]: This is copied from the help text of `migrate`.

File: cms/test_data/management/commands/delete_test_data.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        for model in models:
+            if class_is_indexed(model):
+                post_save.connect(post_save_signal_handler, sender=model)
+
+            if ReferenceIndex.model_is_indexable(model):
+                post_save.connect(update_reference_index_on_save, sender=model)
+
+
+class Command(BaseCommand):
+    help = "Delete all random test data"
```
Comment (kevinhowbrook) [inline cms/test_data/management/commands/delete_test_data.py]:
```suggestion
    help = "Delete all random test data generate by create_test_data"
```

File: cms/test_data/management/commands/create_test_data.py
Comment (RealOrangeOne) [reply cms/test_data/management/commands/create_test_data.py:1]: The problem is you need to be able to pass in arguments to `create_test_data`, which isn't supported when calling through a `Makefile` (easily). Leaving it as a management command to run is probably fine, and not much extra. If we commit a standard config, then absolutely it should be.

File: cms/test_data/management/commands/delete_test_data.py (hunk 5 above)
Comment (RealOrangeOne) [reply cms/test_data/management/commands/delete_test_data.py]: https://github.com/ONSdigital/dis-wagtail/pull/445/commits/6be15754a7778b6a75afb9c50ad413d70d128c1e

File: cms/test_data/tests/test_commands.py (hunk 3 above)
Comment (RealOrangeOne) [reply cms/test_data/tests/test_commands.py]: D'oh! https://github.com/ONSdigital/dis-wagtail/pull/445/commits/36b035cfe11cbf871be47c012b15c984b2b9c848

File: cms/test_data/management/commands/create_test_data.py (hunk 2 above)
Comment (RealOrangeOne) [reply cms/test_data/management/commands/create_test_data.py]: No, no it doesn't :facepalm: https://github.com/ONSdigital/dis-wagtail/pull/445/commits/657ac3a39aedbcde89020dde311c8fb274b14004

File: cms/test_data/config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (RealOrangeOne) [reply cms/test_data/config.py]: Oh right :facepalm: https://github.com/ONSdigital/dis-wagtail/pull/445/commits/a3a1f5cf869eff12fd087465f036371bfa20a94e

File: cms/test_data/management/commands/delete_test_data.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        self.stdout.write("Deleting data...", self.style.NOTICE)
+
+        with disable_signals(list(collector.data.keys())), transaction.atomic():
+            for model, instances in collector.data.items():
+                # Use queryset delete methods to use any customized behaviour
+                model._default_manager.filter(pk__in=[instance.pk for instance in instances]).delete()  # pylint: disable=protected-access
+
+        # NB: Because the topic tree hides the root node in a number core method, it gets corrupted
+        # during deletion. Manually repair the tree afterwards.
+        Topic.fix_tree()
```
Comment (BJacksonONS) [inline cms/test_data/management/commands/delete_test_data.py]: Think most of what I've noticed has already been addressed, but was wondering about this. Might be something to follow up on later rather than do here.

Would we be better off doing something to keep the tree clean and avoiding `fix_tree()`? If this is potentially being run against existing environments with data, this might scale badly as I think it rebuilds everything. Maybe it's not a big issue though.

Comment (BJacksonONS) [reply cms/test_data/management/commands/delete_test_data.py]: Could use a second manager or flag in `get_queryset` to toggle the custom behaviour.

Comment (RealOrangeOne) [reply cms/test_data/management/commands/delete_test_data.py]: The problem is that the root node is hidden in `get_queryset`. I tried removing that, but it broke a lot of other behaviour. Fixing the tree like this felt like the easiest solution. Especially since it'll only be "broken" for a small amount of time, only affect topics, and won't ever be run in an environment where the data matters.

With that said, the comment is clearly nonsense, so I'll tweak that.

Comment (RealOrangeOne) [reply cms/test_data/management/commands/delete_test_data.py]: I've moved the tree fixing inside the delete transaction, which should prevent the "broken" tree from being seen by other connections.

https://github.com/ONSdigital/dis-wagtail/pull/445/commits/436b3997e870169100c41fdbdc3440d7af942603

Comment (BJacksonONS) [reply cms/test_data/management/commands/delete_test_data.py]: I was wondering if we could do something like

```python
from asgiref.local import Local

_disable_topic_root_filtering = Local()

class TopicManager(MP_NodeManager):
    def get_queryset(self) -> TreeQuerySet:
        """Filter out the dummy root topic from all querysets unless disabled."""
        qs = TreeQuerySet(self.model).order_by("path")
        if getattr(_disable_topic_root_filtering, "value", False):
            return qs
        return qs.filter(depth__gt=1)

class Topic():
    ....
    @classmethod
    @contextmanager
    def disable_root_filtering(cls) -> Generator[None, None, None]:
        previous = getattr(_disable_topic_root_filtering, "value", False)
        _disable_topic_root_filtering.value = True
        try:
            yield
        finally:
            _disable_topic_root_filtering.value = previous
```

Then when we're doing the deletions we can just use the context manager

```python
with disable_signals(list(collector.data.keys())), transaction.atomic(), Topic.disable_root_filtering():
```

Should work correctly both in a standalone cron job or if it was called in a server process.

File: cms/taxonomy/tests/factories.py [hunk 7]
```diff
@@ -1,20 +1,32 @@
...
         model = Topic
-        strategy = factory.BUILD_STRATEGY  # To prevent Factory from trying to save the node
+        django_get_or_create = ("id", "parent")
 
-    id: str = faker.Faker("random_number", digits=50, fix_len=True)  # Use a very long ID for uniqueness
-    slug: str = faker.Faker("slug")
-    title: str = faker.Faker("sentence", nb_words=3)
-    description: str = faker.Faker("sentence", nb_words=10)
+    parent = factory.LazyFunction(Topic.objects.root_topic)
+    id = factory.Faker("random_number", digits=50, fix_len=True)  # Use a very long ID for uniqueness
```
Comment (MaciekBaron) [inline cms/taxonomy/tests/factories.py:13]: Question: Why not `factory.Faker("uuid4")`, would that work?

Comment (RealOrangeOne) [reply cms/taxonomy/tests/factories.py:13]: That would also work, but this hasn't been changed from how it was before

File: cms/test_data/management/commands/delete_test_data.py (hunk 6 above)
Comment (RealOrangeOne) [reply cms/test_data/management/commands/delete_test_data.py]: That should work, and feels less risky than a broken tree. I'll see if it works.

Comment (RealOrangeOne) [reply cms/test_data/management/commands/delete_test_data.py]: It seems even that results in a broken tree - the root node has `numchild` which doesn't match its actual children.

Comment (BJacksonONS) [reply cms/test_data/management/commands/delete_test_data.py]: Are we creating items under it while the root is hidden? Maybe there are some writes we're doing that could also be wrapped in the context manager that would give a correct `numchild`?

Comment (RealOrangeOne) [reply cms/test_data/management/commands/delete_test_data.py]: I tried wrapping the entire command in the decorator - same issue.

Comment (BJacksonONS) [reply cms/test_data/management/commands/delete_test_data.py]: Fair, the change in https://github.com/ONSdigital/dis-wagtail/commit/436b3997e870169100c41fdbdc3440d7af942603 probably makes the most sense then.

File: pyproject.toml [hunk 8]
```diff
@@ -47,6 +47,8 @@ wagtailtables = "^0.2.2"
 pyjwt = "^2.13.0"
 cryptography = "^48.0.1"
 django-pglock = "^1.8.0"
+wagtail-factories = "^4.2.1"
```
Comment (BJacksonONS) [inline pyproject.toml]: We still have `wagtail-factories` in dev deps pinned at a higher version ("^4.3.0"). Should clean that up and maybe bump this one to match if we're going to have it in the main group now.

**Comment by BJacksonONS** [comment]
Was just chatting to @MebinAbraham and thought it would be good to add some kind of env var flag so we can disable the command in prod easily.

Given this promotes some dev dependencies to regular ones and unconditionally installs, would provide some peace of mind against human error when running tests later.

**Comment by BJacksonONS** [comment]
Currently getting 

```
django.db.utils.IntegrityError: duplicate key value violates unique constraint "taxonomy_topic_path_key"
DETAIL:  Key (path)=(00010001) already exists.
```
When running this against a freshly created db migrated with `make dev-init`, trying to determine why.

**Comment by BJacksonONS** [comment]
> Currently getting
> 
> ```
> django.db.utils.IntegrityError: duplicate key value violates unique constraint "taxonomy_topic_path_key"
> DETAIL:  Key (path)=(00010001) already exists.
> ```
> 
> When running this against a freshly created db migrated with `make dev-init`, trying to determine why.

I think `make load-topics` is leaving treebeard in a broken state, the create command works fine if I run against a fresh db that has only had migrations run and no topic load.

This needs to run against other environments that have topics loaded, this might be an issue then if we aren't 100% on the cause.

File: cms/test_data/management/commands/create_test_data.py [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+        return created
+
+    def create_from_factory(self, factory_class: type[Factory], factory_kwargs: dict) -> Model:
+        """Create a model instance from a given factory."""
+        return self.create_batch_from_factory(factory_class, 1, factory_kwargs)[0]
+
+    def random_model(self, model: type[Model]) -> Model:
+        """Select a random previously-generated model."""
+        try:
+            return self.faker.random_element(self.model_registry[model])
```
Comment (BJacksonONS) [inline cms/test_data/management/commands/create_test_data.py]: We get an error on this line if any of the model configs have a count of "0".

File: cms/test_data/management/commands/create_test_data.py [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+    try:
+        return TestDataConfig.model_validate_json(config_path.read_text())
+    except ValidationError as e:
+        raise ArgumentTypeError(str(e)) from e
+
+
+class Command(BaseCommand):
+    help = "Create random test data"
+
+    def add_arguments(self, parser: ArgumentParser) -> None:
```
Comment (BJacksonONS) [inline cms/test_data/management/commands/create_test_data.py:163]: An edge case that probably won't matter, but you can pass a blank seed var to get a "None" seed, which means no user input can match and the command fails.

Image

**Comment by RealOrangeOne** [comment]
>  thought it would be good to add some kind of env var flag so we can disable the command in prod easily

It might be, although there are plenty of other commands which cause issues which aren't disabled - `reset_db` for example.

File: cms/test_data/management/commands/create_test_data.py (hunk 9 above)
Comment (RealOrangeOne) [reply cms/test_data/management/commands/create_test_data.py]: https://github.com/ONSdigital/dis-wagtail/pull/445/commits/a4cf296cc86ec799802ee4ff7f45ac36a9839693

File: cms/test_data/management/commands/create_test_data.py (hunk 10 above)
Comment (RealOrangeOne) [reply cms/test_data/management/commands/create_test_data.py:163]: https://github.com/ONSdigital/dis-wagtail/pull/445/commits/e9d0a37cdcfb29613d4d24cb18234ff1a3d663d9

File: pyproject.toml (hunk 8 above)
Comment (RealOrangeOne) [reply pyproject.toml]: https://github.com/ONSdigital/dis-wagtail/pull/445/commits/97c1f9ed5dde7e646996e57340fb0e27d0159a7a

File: cms/test_data/random.py [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+import factory
+import factory.random
+
+
+def get_default_locale() -> str:
+    return str(factory.Faker._DEFAULT_LOCALE)  # pylint: disable=protected-access
+
+
+def _set_default_locale(locale: str) -> None:
+    factory.Faker._DEFAULT_LOCALE = locale  # pylint: disable=protected-access
```
Comment (MaciekBaron) [inline cms/test_data/random.py:13]: **Non-blocking, minor:** Maybe we could add `faker` to the name (`get_default_faker_locale` / `_set_default_faker_locale`) just for better clarity? Code completion might use this by accident for other purposes.

File: cms/test_data/constants.py [hunk 12]
```diff
@@ -0,0 +1 @@
+SEEDED_DATA_PREFIX = "Z-RANDOM "
```
Comment (MaciekBaron) [inline cms/test_data/constants.py]: **Non-blocking, minor:** Perhaps we should move this to settings so it is configurable? In case anyone has a different preference for the generated data.

**Review by MaciekBaron — APPROVED** [review]
LGTM, just two non-blocking comments.

File: cms/test_data/random.py (hunk 11 above)
Comment (BJacksonONS) [reply cms/test_data/random.py:13]: https://github.com/ONSdigital/dis-wagtail/pull/445/commits/27a9e1ff350ec57fa653f380ec63f00e616a6c23

File: cms/test_data/constants.py (hunk 12 above)
Comment (BJacksonONS) [reply cms/test_data/constants.py]: https://github.com/ONSdigital/dis-wagtail/pull/445/commits/6aa3e6c9c94b909d076d9be5fa93e2f5e54853b7

**Review by kevinhowbrook — APPROVED** [review]
lgtm, only spot is that we could add `DEFAULT_LOCALE = "en_GB"` but it's not a blocker and potentially not worth it at this point.
Great work picking this up and getting it over the line 🥳

**Comment by MebinAbraham** [comment]
Trialling Copilot reviews. Will add instructions and AGENTS.md in the near future.

File: cms/topics/tests/factories.py [hunk 13]
```diff
@@ -5,16 +5,47 @@
...
+
+
 class TopicPageFactory(wagtail_factories.PageFactory):
     """Factory for TopicPage."""
 
     class Meta:
         model = TopicPage
+        django_get_or_create = ("title", "parent")
 
     parent = factory.LazyFunction(lambda: HomePage.objects.first())  # pylint: disable=unnecessary-lambda
```
[the remark this answers is not in the document]
Comment (BJacksonONS) [reply cms/topics/tests/factories.py:50]: Addressed here

https://github.com/ONSdigital/dis-wagtail/pull/445/commits/9a0a0047574d7d6394330991b04fda9dff21c229

File: cms/test_data/management/commands/create_test_data.py [hunk 14]
```diff
@@ -0,0 +1,196 @@
...
+        self.title_factory = factory.LazyFunction(self.random_title)  # type: ignore[no-untyped-call]
+
+        self.root_page = Site.objects.get(is_default_site=True).root_page
+
+        self.model_registry: dict[type[Model], list[Model]] = {}
+
+        self.get_config_count = partial(self.config.get_count, faker=self.faker)
+
+    def random_title(self) -> str:
+        return settings.CMS_TEST_DATA_PREFIX + self.faker.sentence(nb_words=3)
```
[the remark this answers is not in the document]
Comment (BJacksonONS) [reply cms/test_data/management/commands/create_test_data.py]: Addressed here

https://github.com/ONSdigital/dis-wagtail/pull/445/commits/005028b8d5b04340ecc55af31f3d8216befb6605

File: cms/taxonomy/models.py [hunk 15]
```diff
@@ -17,15 +18,16 @@
...
 
 
-class TopicManager(models.Manager):
-    def get_queryset(self) -> QuerySet:
+class TopicManager(MP_NodeManager):
+    def get_queryset(self) -> TreeQuerySet:
         """Filter out the dummy root topic from all querysets."""
-        return super().get_queryset().filter(depth__gt=1)
+        # Reuse Wagtail's custom tree QuerySet for helpful utils
+        return TreeQuerySet(self.model).order_by("path").filter(depth__gt=1)
```
[the remark this answers is not in the document]
Comment (BJacksonONS) [reply cms/taxonomy/models.py:26]: Not sure this has much impact, have fixed the _db and _hints via kwargs on TreeQuerySet.

https://github.com/ONSdigital/dis-wagtail/pull/445/commits/ba5096619f2ea23e948a025565828d9ef2d105bc

**Comment by BJacksonONS** [comment]
Trying a rereview from copilot. Will see if it resolves it's own previous comments.

File: cms/test_data/management/commands/delete_test_data.py [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
+            action="store_true",
+            help="Just show the models to delete - don't delete anything",
+        )
+
+    def _get_lookups(self, model: type[Model]) -> Q:
+        """Query any field which might contain the prefix."""
+        lookups = Q()
+        for field in model._meta.fields:
+            if field.is_relation or not isinstance(field, CharField | TextField):
+                continue
```
[the remark this answers is not in the document]
Comment (BJacksonONS) [reply cms/test_data/management/commands/delete_test_data.py:38]: This is valid as of python 3.10 and we're on 3.14. Would have caused failures in tests if this was an issue.

**Review by MebinAbraham — COMMENTED** [review]
We need to document these. Let's add instructions to README.md.

Can we also follow up with a doc under docs/ for this.

File: cms/test_data/management/commands/create_test_data.py [hunk 17]
```diff
@@ -0,0 +1,198 @@
+import itertools
```
Comment (MebinAbraham) [inline cms/test_data/management/commands/create_test_data.py:1]: Let's add these to Makefile to easy invocation.

**Review by sanjeevz3009 — APPROVED** [review]
LGTM!

File: cms/test_data/constants.py (hunk 12 above)
Comment (sanjeevz3009) [reply cms/test_data/constants.py]: Resolving on behalf!

File: cms/test_data/random.py (hunk 11 above)
Comment (sanjeevz3009) [reply cms/test_data/random.py:13]: Resolving on behalf!

File: cms/taxonomy/tests/factories.py (hunk 7 above)
Comment (sanjeevz3009) [reply cms/taxonomy/tests/factories.py:13]: Resolving on behalf!

**Comment by sanjeevz3009** [comment]
Apart from @MebinAbraham comment here: https://github.com/ONSdigital/dis-wagtail/pull/445#discussion_r3719830654, the branch needs updating with main, and the `poetry.lock` file merge conflicts need resolving.

This PR is good to go in.

File: .pylintrc [hunk 18]
```diff
@@ -431,6 +431,7 @@ disable=raw-checker-failed,
         missing-function-docstring,
         fixme,
         too-many-public-methods,
+        cyclic-import,
```
Comment (MebinAbraham) [inline .pylintrc:434]: 👀 👀 👀 👀 👀 👀 😄

Comment (BJacksonONS) [reply .pylintrc:434]: That's from main, was introduced here

https://github.com/ONSdigital/dis-wagtail/pull/676/changes#diff-ba293004ae8f18975051e825bd8d5c70137a8b5d42d16e0f7c9354aa634c446cR434

File: cms/test_data/management/commands/create_test_data.py (hunk 17 above)
Comment (BJacksonONS) [reply cms/test_data/management/commands/create_test_data.py:1]: https://github.com/ONSdigital/dis-wagtail/pull/445/commits/a85e58ebd2e79f312583395beb52bdc42af26996

File: Makefile [hunk 19]
```diff
@@ -115,6 +115,25 @@ load-design-system-templates:  ## Load the design system templates
...
+.PHONY: delete-test-data-dry-run
+delete-test-data-dry-run:  ## Dry run for deleting test data
+	poetry run python ./manage.py delete_test_data --dry-run
+
+.PHONY: delete-test-data
+delete-test-data:  ## Delete test data. Use NOINPUT=1 to skip confirmation prompt.
+	poetry run python ./manage.py delete_test_data $(if $(NOINPUT), --noinput)
+
+.PHONY: show-default-test-data-config
+delete-test-data:  ## Show default config. Use SCHEMA=1 to show JSON schema.
```
Comment (MebinAbraham) [inline Makefile]: How about prefixing `test-data` for easier completions? Thoughts?

Comment (MebinAbraham) [inline Makefile]: Wrong command key, should be `show-default-test-data-config`.

Comment (BJacksonONS) [reply Makefile]: Ooops, silly copy paste 🤦

https://github.com/ONSdigital/dis-wagtail/pull/445/commits/5739bbd7c7962dbc677ee3dfa1bd383d492db10a

Comment (BJacksonONS) [reply Makefile]: Yeah I like that

https://github.com/ONSdigital/dis-wagtail/pull/445/commits/5739bbd7c7962dbc677ee3dfa1bd383d492db10a

Comment (BJacksonONS) [reply Makefile]: My only other thought on that is maybe making it harder to accidentally hit the delete command rather than the dry run, but it requires user input to follow through by default anyway.

Comment (MebinAbraham) [reply Makefile]: Ya, since it needs follow-up prompts, I think that's okay.

**Review by MebinAbraham — APPROVED** [review]

File: .pylintrc (hunk 18 above)
Comment (MebinAbraham) [reply .pylintrc:434]: We'll come back to this.

File: Makefile [hunk 20]
```diff
@@ -115,6 +115,25 @@ load-design-system-templates:  ## Load the design system templates
 load-topics:  ## Load our fixture of taxonomy topics
 	poetry run python ./manage.py loaddata cms/taxonomy/fixtures/topics.json
 
+.PHONY: test-data-create
+test-data-create:  ## Seed test data
+	poetry run python ./manage.py create_test_data \
+		$(if $(SEED), --seed $(SEED)) \
+		$(if $(CONFIG), --config $(CONFIG)) \
+		$(if $(NOINPUT), --config) \
```
Comment (MebinAbraham) [inline Makefile]:
```suggestion
		$(if $(NOINPUT), --noinput) \
```

**Review by sanjeevz3009 — APPROVED** [review]
LGTM!

**Review by sanjeevz3009 — APPROVED** [review]
